STL_Extension/include/CGAL/Handle.h

```diff
@@ -122,7 +122,7 @@ class Handle
     int
     refs()  const noexcept { return PTR->count.load(std::memory_order_relaxed); }
 
-    Id_type id() const noexcept { return PTR - static_cast<Rep*>(0); }
+    Id_type id() const noexcept { return static_cast<Id_type>(reinterpret_cast<std::intptr_t>(static_cast<void*>(PTR)) / sizeof(Rep)); }
```

Comment:
This conflicts with another fix:
```
commit 4d797b55c494f48dc1ffcfe558cd17dd3a29e9ef
Author: Sébastien Loriot <sebastien.loriot@cgal.org>
Date:   Thu Nov 17 19:03:30 2022 +0100

    try working around a warning

diff --git a/STL_Extension/include/CGAL/Handle.h b/STL_Extension/include/CGAL/Handle.h
index 4fa3a17286b..07fb7748e21 100644
--- a/STL_Extension/include/CGAL/Handle.h
+++ b/STL_Extension/include/CGAL/Handle.h
@@ -122,7 +122,7 @@ class Handle
     int
     refs()  const noexcept { return PTR->count.load(std::memory_order_relaxed); }
 
-    Id_type id() const noexcept { return PTR - static_cast<Rep*>(0); }
+    Id_type id() const noexcept { return std::distance(static_cast<Rep*>(0), PTR); }
 
     bool identical(const Handle& h) const noexcept { return PTR == h.PTR; }
``` 
I guess you should merge `master` and fix the conflict.

Reply:
I'm not sure to understand why you divide by `sizeof(Rep)`, this will change the current behavior, right?

Reply:
Pointer arithmetic is a nasty thing. In cpprer

Reply:
[In cppreference.com, we can read](https://en.cppreference.com/w/cpp/language/operator_arithmetic#Additive_operators):
> - If the pointer `P` points to the ith element of an array, and the pointer `Q` points at the jth element of the same array, the expression `P - Q` has the value `i - j`, if the value fits in [std::ptrdiff_t](https://en.cppreference.com/w/cpp/types/ptrdiff_t). Both operands must point to the elements of the same array (or one past the end), otherwise the behavior is undefined. If the result does not fit in [std::ptrdiff_t](https://en.cppreference.com/w/cpp/types/ptrdiff_t), the behavior is undefined.

That paragraph sums up all the issue:
  - `PTR - nullptr` is undefined behavior, because `PTR` and `nullptr` do not point to elements of an array.
  - the use of `std::distance` is the same undefined behavior,
  - `P - Q` has the value `i - j`, and that computation involves a division by `sizeof(Rep)`. For example, if `Q = P + 1`, points to a the element just next to the element pointed by `P`, in array of `int`, then
    - `Q - P  == 1`, but
    - `reinterpret_cast<std::intptr_t>(Q) - reinterpret_cast<std::intptr_t>(P) == sizeof(int)`.